### PR TITLE
V4

### PR DESCRIPTION
**Pull request by ardalis**
Version 4 Update

### CODE REVIEW

**Comment by ardalis** [comment]
@maxkoshevoi Can you take a look before I merge this?

File: sample/SampleEndpointApp/Endpoints/Authors/Get.cs [hunk 1]
```diff
@@ -1,36 +1,39 @@
-﻿using Ardalis.ApiEndpoints;
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Ardalis.ApiEndpoints;
 using AutoMapper;
 using Microsoft.AspNetCore.Mvc;
 using SampleEndpointApp.DomainModel;
 
-namespace SampleEndpointApp.Endpoints.Authors;
```
Comment (maxkoshevoi) [inline sample/SampleEndpointApp/Endpoints/Authors/Get.cs:6]: Why have you reverted to block-scoped namespaces here? Are they also block-scoped in the rest of the files (cannot remember)

File: src/ApiEndpoints.NSwag/Ardalis.ApiEndpoints.NSwag.csproj [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (maxkoshevoi) [inline src/ApiEndpoints.NSwag/Ardalis.ApiEndpoints.NSwag.csproj:1]: There's already a https://github.com/ardalis/ApiEndpoints/blob/main/src/Ardalis.ApiEndpoints.NSwag/Ardalis.ApiEndpoints.NSwag.csproj why is this project file needed? 
If you want to rename `Ardalis.ApiEndpoints.NSwag` folder to `ApiEndpoints.NSwag`, you need to also delete files from `Ardalis.ApiEndpoints.NSwag`

File: sample/SampleEndpointApp/Endpoints/Authors/Get.cs (hunk 1 above)
Comment (ardalis) [reply sample/SampleEndpointApp/Endpoints/Authors/Get.cs:6]: Probably chose poorly on the massive merge conflict I was dealing with. Sigh. I'll try to sort it out.

File: src/ApiEndpoints.NSwag/Extensions/NSwagSwaggerGeneratorSettingsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
```
Comment (maxkoshevoi) [inline src/ApiEndpoints.NSwag/Extensions/NSwagSwaggerGeneratorSettingsExtensions.cs:1]: Same for this file

**Review by maxkoshevoi — CHANGES_REQUESTED** [review]
Looks good, I would also add a line in readme that says that user can inherit directly from `BaseEndpoint` to remove restriction on action schema, and retain validation that controller has only one action

File: sample/SampleEndpointApp/Endpoints/Authors/Get.cs (hunk 1 above)
Comment (maxkoshevoi) [reply sample/SampleEndpointApp/Endpoints/Authors/Get.cs:6]: At least it's only one file. What merge conflict BTW? I thought everything is already in `main`

**Pull request closed without merging**